ddl/db_test.go

```diff
@@ -3096,6 +3096,59 @@ func (s *testDBSuite) TestPartitionUniqueKeyNeedAllFieldsInPf(c *C) {
 	);`
 	s.testErrorCode(c, sql8, tmysql.ErrUniqueKeyNeedAllFieldsInPf)
 }
+func (s *testDBSuite) TestPartitionDropIndex(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("set @@tidb_enable_table_partition = 1")
+	tk.MustExec("drop table if exists tr;")
+	tk.MustExec(`create table tr(
+		id int, name varchar(50), 
+		purchased date
+	)
+	partition by range( year(purchased) ) (
+    	partition p0 values less than (1990),
+    	partition p1 values less than (1995),
+    	partition p2 values less than (2000),
+    	partition p3 values less than (2005),
+    	partition p4 values less than (2010),
+    	partition p5 values less than (2015)
+   	);`)
+	tk.MustExec(` insert into tr values
+    (1, 'desk organiser', '2003-10-15'),
+    (2, 'alarm clock', '1997-11-05'),
+    (3, 'chair', '2009-03-10'),
+    (4, 'bookcase', '1989-01-10'),
+    (5, 'exercise bike', '2014-05-09'),
+    (6, 'sofa', '1987-06-05'),
+    (7, 'espresso maker', '2011-11-22'),
+    (8, 'aquarium', '1992-08-04'),
+    (9, 'study desk', '2006-09-16'),
+    (10, 'lava lamp', '1998-12-25');`)
+
+	tk.MustExec("create index idx1 on tr (purchased)")
+	tk.MustQuery("select count(purchased) from tr use index(idx1)").Check(testkit.Rows("10"))
+	tk.MustExec("admin check table tr")
+	// Wait for add index done.
+	time.Sleep(waitForCleanDataInterval)
```

Comment:
There is no need to wait.
 `create index idx1 on tr (purchased)`  blocks for a while, as long as it returns, add index will be done.